spec/controllers/admin/tool_rentings_controller_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'rails_helper'
+
+RSpec.describe Admin::ToolRentingsController, type: :controller do
+  let(:user) { create(:user) }
+
+  allow_user_to :manage, ToolRenting
+
+  before(:each) do
+    allow(controller).to receive(:current_user).and_return(user)
+  end
+
+  describe 'POST #create' do
+    it 'valid parameters' do
+      tool = create(:tool)
+      attributes = attributes_for(:tool_renting, tool: tool)
+
+      lambda do
+        post :create, tool_id: tool, tool_renting: attributes
+      end.should change(ToolRenting, :count).by(1)
+
+      response.should redirect_to(admin_tool_path(tool))
+      ToolRenting.last.renter.should eq(attributes[:renter])
+    end
+
+    it 'invalid parameters' do
+      tool = create(:tool)
+      attributes = attributes_for(:tool_renting, renter: nil, tool: tool)
+
+      lambda do
+        post :create, tool_id: tool, tool_renting: attributes
+      end.should change(ToolRenting, :count).by(0)
+
+      response.status.should eq(422)
+      response.should render_template(:new)
+    end
+  end
+
+  describe 'GET #new' do
+    it 'loads new view properly' do
+      tool = create(:tool)
+      rent = create(:tool_renting, tool: tool)
+      get :new, id: rent.to_param, tool_id: tool.to_param
+      response.status.should eq(200)
+    end
+  end
+
+  describe 'GET #edit' do
+    it 'loads edit view properly' do
+      tool = create(:tool)
+      rent = create(:tool_renting, tool: tool)
+      get :edit, id: rent.to_param, tool_id: tool.to_param
+      response.status.should eq(200)
+    end
+  end
+
+  describe 'PATCH #update' do
+    it 'valid parameters' do
+      tool = create(:tool)
+      rent = create(:tool_renting, tool: tool)
```

Comment:
You can pass arguments to FactoryGirl as well:
`create(:tool_renting, tool: tool, renter: 'Rotharen')`